### PR TITLE
Cache problems #273

### PR DESCRIPTION
**Pull request by ammaratef45**
Solved problems in #273
1- cache is accessible from any module by importing it.
2- Redis hostname is being set from config.
3- removed cache from the base template.

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#294](https://github.com/timelesslounge/timelessis/issues/294) is now in scope, role is `REV`

**Comment by 0crat** [comment]
This pull request [#294](https://github.com/timelesslounge/timelessis/issues/294) is assigned to @vryazanov[/z](https://www.0crat.com/u/vryazanov), here is [why](https://www.0crat.com/footprint/CF9LG1YCA/678adf77-98f4-4a7c-ac4b-3efb15d7887b); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @emilianodellacasa[/z](https://www.0crat.com/u/emilianodellacasa) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

File: timeless/__init__.py [hunk 1]
```diff
@@ -16,7 +16,13 @@ def create_app(config):
     """Creates a new Timeless webapp given a config class"""
     app = Flask(__name__, instance_relative_config=True)
     app.config.from_object(config)
-    cache = Cache(app, config={"CACHE_TYPE": "redis"})
+    CACHE.init_app(
```
Comment (vryazanov) [inline timeless/__init__.py]: @ammaratef45 can we just pass `config`, like `CACHE.init_app(app, config=config)`? And define `CACHE_TYPE` and `CACHE_REDIS_HOST` in config object

File: config.py [hunk 2]
```diff
@@ -39,6 +41,7 @@ class DevelopmentConfig(Config):
         "postgresql://timeless_user:timeless_pwd@localhost/timelessdb_dev")
     RESULT_BACKEND = "redis://localhost:6379"
     BROKER_URL = "redis://localhost:6379"
+    REDIS_HOST = "localhost"
```
Comment (vryazanov) [inline config.py]: @ammaratef45 all these vars are looking at `redis`:
```python
    RESULT_BACKEND = "redis://localhost:6379"
    BROKER_URL = "redis://localhost:6379"
    REDIS_HOST = "localhost"
```
so I suggest to do the following:
```python
    REDIS_HOST =  "redis://localhost:6379"
    RESULT_BACKEND = REDIS_HOST
    BROKER_URL = REDIS_HOST
```

wdyt?

Comment (ammaratef45) [reply config.py]: @vryazanov I think you are right

**Comment by ammaratef45** [comment]
@vryazanov please take a look

File: timeless/__init__.py [hunk 3]
```diff
@@ -7,18 +7,22 @@
...
     app = Flask(__name__, instance_relative_config=True)
     app.config.from_object(config)
-    cache.init_app(app, config={"CACHE_TYPE": "redis"})
+    redis_config = {
+        "CACHE_TYPE": "redis",
+        "CACHE_REDIS_URL": app.config.get("REDIS_HOST")
+    }
+    CACHE.init_app(
+        app,
+        config=redis_config
```
Comment (vryazanov) [inline timeless/__init__.py]: @ammaratef45 I expected a bit different, see for example https://github.com/gofynd/flask-full/blob/master/app/__init__.py#L47

We can implement something like:
```python
class StagingConfig(Config):
    REDIS_HOST = "redis://localhost:6379"
    RESULT_BACKEND = REDIS_HOST
    BROKER_URL = REDIS_HOST
    CACHE_SETTINGS = {
        "CACHE_TYPE": "redis",
        "CACHE_REDIS_URL": REDIS_HOST
    }
```
and then pass it to `Cache.init_app`: `CACHE.init_app(app, config=config.CACHE_SETTINGS)`.
So this way all settings will be in one place

Comment (ammaratef45) [reply timeless/__init__.py]: @vryazanov looks better 👍

**Comment by ammaratef45** [comment]
@vryazanov check now

**Comment by vryazanov** [comment]
@rultor merge

**Comment by emilianodellacasa** [comment]
@ammaratef45 please check codeclimate

**Comment by ammaratef45** [comment]
@emilianodellacasa I don't know a way to solve Similar blocks of code for the config file

**Comment by ammaratef45** [comment]
@emilianodellacasa I marked it as a wontfix is this fine?

**Comment by emilianodellacasa** [comment]
@rultor merge

**Comment by 0crat** [comment]
Job was finished in 25 hours, bonus for fast delivery is possible (see [§36](http://www.zerocracy.com/policy.html#36))

**Comment by 0crat** [comment]
Job [#294](https://github.com/timelesslounge/timelessis/issues/294) is not in the agenda of @vryazanov[/z](https://www.0crat.com/u/vryazanov), can't inspect

**Comment by 0crat** [comment]
The job [#294](https://github.com/timelesslounge/timelessis/issues/294) is now out of scope

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://www.zerocracy.com/policy.html#28): +10 point(s) just awarded to @emilianodellacasa[/z](https://www.0crat.com/u/emilianodellacasa)

**Comment by vryazanov** [comment]
@0crat status

**Comment by 0crat** [comment]
> @0crat status ([here](https://github.com/timelesslounge/timelessis/issues/294#issuecomment-465511737))

@vryazanov This is what I know about this job in [CF9LG1YCA](https://www.0crat.com/p/CF9LG1YCA), as in [§32](http://www.zerocracy.com/policy.html#32):

  * The job [#294](https://github.com/timelesslounge/timelessis/issues/294) is not in scope
  * The job is waiting quality review verdict by @ypshenychka[/z](https://www.0crat.com/u/ypshenychka) for 31min
  * @vryazanov[/z](https://www.0crat.com/u/vryazanov) will receive 20 minutes after quality review
  * Job [footprint](https://www.0crat.com/footprint/CF9LG1YCA?q=%7Bjob%3A%27gh%3Atimelesslounge%2Ftimelessis%23294%27%2Ctype%3A%7B%24not%3A%2F%5ENotify%2F%7D%7D) (restricted area)

**Comment by ypshenychka** [comment]
@0crat quality good

**Comment by 0crat** [comment]
Quality review completed: +8 point(s) just awarded to @ypshenychka[/z](https://www.0crat.com/u/ypshenychka)

**Comment by 0crat** [comment]
Order was finished, quality is "good": +25 point(s) just awarded to @vryazanov[/z](https://www.0crat.com/u/vryazanov)

**Pull request merged**